### PR TITLE
refactor: Migrate from JUnit 4 to JUnit 5 (#2545)

### PR DESCRIPTION
**Pull request by nine03**
### Purpose

refactor #2545 
I'm working on migrating to junit5 in TestBooleanCounterProcessor, but the code has changed so much that I'm worried that I'm writing the code correctly, and I want to fix the errors through code review.

### Remarks

PR introduces (a) breaking change(s): no

PR introduces (a) deprecation(s): no

### CODE REVIEW

File: streampipes-extensions/streampipes-processors-transformation-jvm/src/test/java/org/apache/streampipes/processors/transformation/jvm/processor/booloperator/counter/TestBooleanCounterProcessor.java [hunk 1]
```diff
@@ -46,50 +48,38 @@
...
+  ){
 
-  @org.junit.runners.Parameterized.Parameter(3)
-  public Integer expectedBooleanCount;
-
-  @Test
-  public void testBooleanCounter() {
     BooleanCounterProcessor booleanCounter = new BooleanCounterProcessor();
-    DataProcessorDescription originalGraph = booleanCounter.declareConfig().getDescription();
+//    DataProcessorDescription originalGraph = booleanCounter.declareConfig().getDescription();
```
Comment (nine03) [inline streampipes-extensions/streampipes-processors-transformation-jvm/src/test/java/org/apache/streampipes/processors/transformation/jvm/processor/booloperator/counter/TestBooleanCounterProcessor.java:80]:
```
DataProcessorDescription originalGraph = booleanCounter.declareConfig().getDescription();
```

Initially it was written like this, but after junit5 migration it didn't work.

```
IDataProcessorConfiguration config = booleanCounter.declareConfig();
DataProcessorDescription originalGraph = config.getDescription();
```

and I got the error.

File: streampipes-extensions/streampipes-processors-transformation-jvm/src/test/java/org/apache/streampipes/processors/transformation/jvm/processor/booloperator/counter/TestBooleanCounterProcessor.java [hunk 2]
```diff
@@ -132,15 +122,22 @@ public void disconnect() throws SpRuntimeException {}
     Integer counter = sendEvents(booleanCounter, spOut);
```
Comment (nine03) [inline streampipes-extensions/streampipes-processors-transformation-jvm/src/test/java/org/apache/streampipes/processors/transformation/jvm/processor/booloperator/counter/TestBooleanCounterProcessor.java:122]: I'm getting an error on lines 121-122

```
booleanCounter.onPipelineStarted(params, spOut, null);
Integer counter = sendEvents(booleanCounter, spOut);
```

Params, spOut, booleanCounter. There's an error in this part

File: streampipes-extensions/streampipes-processors-transformation-jvm/src/test/java/org/apache/streampipes/processors/transformation/jvm/processor/booloperator/counter/TestBooleanCounterProcessor.java [hunk 3]
```diff
@@ -132,15 +122,22 @@ public void disconnect() throws SpRuntimeException {}
...
+
+  private Integer sendEvents(BooleanCounterProcessor booleanCounter, SpOutputCollector spOut, List<Boolean> eventBooleans, String invertFieldName) {
     int counter = 0;
-    List<Event> events = makeEvents();
+    List<Event> events = makeEvents(eventBooleans, invertFieldName);
     for (Event event : events) {
       LOG.info("Sending event with value "
-          + event.getFieldBySelector("s0::" + invertFieldName).getAsPrimitive().getAsBoolean());
+              + event.getFieldBySelector("s0::" + invertFieldName).getAsPrimitive().getAsBoolean());
       booleanCounter.onEvent(event, spOut);
```
Comment (nine03) [inline streampipes-extensions/streampipes-processors-transformation-jvm/src/test/java/org/apache/streampipes/processors/transformation/jvm/processor/booloperator/counter/TestBooleanCounterProcessor.java:141]: There's an error in line 141.

```
booleanCounter.onEvent(event, spOut);
```

There's an error in spOut.

File: streampipes-extensions/streampipes-processors-transformation-jvm/src/test/java/org/apache/streampipes/processors/transformation/jvm/processor/booloperator/counter/TestBooleanCounterProcessor.java (hunk 2 above)
Comment (nine03) [inline streampipes-extensions/streampipes-processors-transformation-jvm/src/test/java/org/apache/streampipes/processors/transformation/jvm/processor/booloperator/counter/TestBooleanCounterProcessor.java:122]: I'm getting an error on lines 121-122

```
booleanCounter.onPipelineStarted(params, spOut, null);
Integer counter = sendEvents(booleanCounter, spOut);
```

Params, spOut, booleanCounter. There's an error in this part.

**Comment by tenthe** [comment]
Hi @nine03,
thank you for your PR.
Can you please update your local git repository to the new `dev` branch. I checked the code out and it looks like it is using junit4 instead of junit5. I could not run it locally.

Yes, you are right, a lot has changed. But now it is also possible to simplify the test code a lot.
Especially how to instantiate the test class `BooleanCounterProcessor`.
Therefore you can have a look at the test of `TestBooleanFilterProcessor`. Here I tried to clean up the code a bit.

Let me know if you have any questions or challenges. I hope this explanation helps a little.

**Comment by nine03** [comment]
> Hi @nine03, thank you for your PR. Can you please update your local git repository to the new `dev` branch. I checked the code out and it looks like it is using junit4 instead of junit5. I could not run it locally.
> 
> Yes, you are right, a lot has changed. But now it is also possible to simplify the test code a lot. Especially how to instantiate the test class `BooleanCounterProcessor`. Therefore you can have a look at the test of `TestBooleanFilterProcessor`. Here I tried to clean up the code a bit.
> 
> Let me know if you have any questions or challenges. I hope this explanation helps a little.

Hi @tenthe,
I have merged the changes from 2545-testbooleancounterprocessor branch into the dev branch as you requested. Please take a look when you have time. Thank you!

**Comment by nine03** [comment]
> Hi @nine03, thank you for your PR. Can you please update your local git repository to the new `dev` branch. I checked the code out and it looks like it is using junit4 instead of junit5. I could not run it locally.
> 
> Yes, you are right, a lot has changed. But now it is also possible to simplify the test code a lot. Especially how to instantiate the test class `BooleanCounterProcessor`. Therefore you can have a look at the test of `TestBooleanFilterProcessor`. Here I tried to clean up the code a bit.
> 
> Let me know if you have any questions or challenges. I hope this explanation helps a little.

Hi @tenthe, 
I have merged the changes from 2545-testbooleancounterprocessor branch into the dev branch as you requested and created a new pull request #2567. Please take a look when you have time. Thank you!

**Pull request closed without merging**